clang/lib/AST/ASTStructuralEquivalence.cpp

```diff
@@ -214,6 +214,16 @@ class StmtComparer {
     return E1->size() == E2->size();
   }
 
+  bool IsStmtEquivalent(const DeclRefExpr *DRE1, const DeclRefExpr *DRE2) {
+    auto *Decl1 = DRE1->getDecl();
+    auto *Decl2 = DRE2->getDecl();
```

Comment:
actually you can make these `const`, and then you won't need the `const_cast`s

Reply:
```cpp
/// Determine structural equivalence of two declarations.
static bool IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
                                     Decl *D1, Decl *D2) {
  // FIXME: Check for known structural equivalences via a callback of some sort.

  D1 = D1->getCanonicalDecl();
  D2 = D2->getCanonicalDecl();
  std::pair<Decl *, Decl *> P{D1, D2};
```
```IsStructurallyEquivalent``` only accept pointer without const qualifier.